### DISCUSSION TITLE
error: authentication error: error during receive response: EOF

### DISCUSSION THREAD

**Discussion by lorbetzki** [spali/go-rscp#5] [Q&A] [answered]
Hi,

da bin ich der erste der ins Forum schreibt und gleich mit einer bestimmt dummen Frage.

habe die letzte Version runtergeladen und mir eine .config erstellt

```
root@lo:/usr/src/go-rscp# cat .config
host 192.168.147.5
user Mail@Domain.net
password meinGeheimesPortalPassword
key meinRSCPKey

```
unter key, steht der Key, den ich an der E3DC Anlage manuell für RSCP vergeben habe. Nun rufe ich anhand des examples etwas auf und bekomme folgende Meldung

```
root@lo:/usr/src/go-rscp# ./e3dc -debug 99 '["INFO_REQ_MAC_ADDRESS"]'
INFO[0000] Connecting to 192.168.147.5:5033
INFO[0000] successfully connected to 192.168.147.5:5033
DEBU[0000] write [{ RSCP_REQ_AUTHENTICATION Container [{ RSCP_AUTHENTICATION_USER CString Mail@Domain.net } { RSCP_AUTHENTICATION_PASSWORD CString ******** }] }]
TRAC[0000] write plain []byte{....}
TRAC[0000] write crypt []byte{....}
INFO[0000] disconnected
error: authentication error: error during receive response: EOF
```

Hab eine S10E mit der Firmware S10_2021_04. Eine idee was ich falsch gemacht haben könnte?

**Comment by spali** [comment, marked as answer]
Hallo
Ist schwer zu sagen... leider ist das RSCP Protokoll nicht sehr gesprächig bei der Authentifikation wenn was schief läuft und bricht manchmal einfach die Kommunikation ab.
Ebenfalls stelle sicher das alle Angaben im `.config` stimmen.
Ev. hilft es den Key nochmals neu zu setzen in der Anlage.

Bist du sicher das du die letzte Version 0.1.0 hast? Die hat noch ein paar kleinere Handling-Verbesserungen bei der Authentifizierung.
Zu mindestens mit neusten Version würden fehlerhafter Username/Passwort mit mit folgender Ausgabe enden:
```
error: authentication failed: AUTH_LEVEL_NO_AUT
```
Und dein Error würde auf ein fehlerhaften Key hindeuten (da dann die Kommunikation komplett abbricht):
```
error: authentication error: error during receive response: EOF
```



**PS: Habe in deinem Log Auszug die Authentifizierung zensiert. Bitte benutze max. `-debug 6`. Dann sieht man alles, aber die Authentifizierung ist automatisch zensiert. Ansonsten könnte jemand deine Login Daten daraus entziffern.**

**Reply by lorbetzki** [reply]
Hi, 
danke fürs editieren. Copy and paste ist nicht immer gut. Zur Version:
```
root@lo:/usr/src/go-rscp# ./e3dc -version
e3dc
----
Source:     https://github.com/spali/go-rscp
Version:    0.1.0
Commit:     74d28ec9e0cd212b248257764a34c6a50d79c1f7
Platform:   linux/amd64
Build Time: 2021-07-31T19:47:21Z

```
quasi frisch aus dem Releases gezogen. Ich werde morgen mal testen den Key nochmal zu ändern.
